embeddings.go

```diff
@@ -113,10 +113,21 @@ type EmbeddingResponse struct {
 	Usage  Usage          `json:"usage"`
 }
 
+type EmbeddingRequestBody interface {
+	// Needs to be of type EmbeddingRequest or EmbeddingRequestTokens
+	ToEmbeddingRequest() BaseEmbeddingRequest
+}
+
+type BaseEmbeddingRequest struct {
+	Input interface{}    `json:"input"`
```

Comment:
Looks good, thank you for the PR! Could you please change `interface{}` to `any`?